File: common/internal/src/main/java/dcp/mc/pstp/config/Ownership.java
package dcp.mc.pstp.config;

import org.jetbrains.annotations.ApiStatus.Internal;
import org.jetbrains.annotations.NotNull;

@Internal
public record Ownership(
        boolean removal,
        boolean adding,
        boolean transfer,
        boolean defaultShears,
        @NotNull String @NotNull [] shears,
        boolean defaultStick,
        @NotNull String @NotNull [] sticks
)
{
    Ownership() {
        this(true, true, false, true, new String[0], true, new String[0]);
    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/Ownership.kt
package dcp.mc.pstp.config
@JvmRecord
data class Ownership(
    val removal: Boolean = true,
    val adding: Boolean = true,
    val transfer: Boolean = false,
    val defaultShears: Boolean = true,
    val extraShears: List<String> = emptyList(),
    val defaultStick: Boolean = true,
    val extraSticks: List<String> = emptyList(),
)